### PR TITLE
gottafixit 🤠

### PR DESCRIPTION
**Pull request by cubancorona**

### CODE REVIEW

File: blockmanager.go [hunk 1]
```diff
@@ -2620,6 +2620,13 @@ func (b *blockManager) handleHeadersMsg(hmsg *headersMsg) {
 					hmsg.peer.Disconnect()
 					return
 				}
+				// Make sure headers being processed are connecting
```
Comment (guggero) [inline blockmanager.go:2623]: nit: can you please add a newline before the comment and after your added block of code? Also, sentences should always be complete, including punctuation.

File: blockmanager.go [hunk 2]
```diff
@@ -2620,6 +2620,13 @@ func (b *blockManager) handleHeadersMsg(hmsg *headersMsg) {
...
+				prevBlockHashInReorganizationList := b.reorgList.Back().Header.BlockHash()
+				if !reorgHeader.PrevBlock.IsEqual(&prevBlockHashInReorganizationList) {
```
Comment (guggero) [inline blockmanager.go:2625]: We try to keep lines at 80 characters max length (using 8 spaces for tabs). Maybe choose a shorter name?

**Review by guggero — COMMENTED** [review]
Thanks for the PR. What's the context here? Did you run into this in the wild?

**Comment by cubancorona** [comment]
> Thanks for the PR. What's the context here? Did you run into this in the wild?

No.  I simulated a block collision (two new valid blocks mined at approx. the same time, and at same height) using real block data (i.e., where an actual block race occurred) and then appended a BS chain to the de facto abandoned fork to show that neutrino would follow my garbage chain...

File: blockmanager.go (hunk 1 above)
Comment (cubancorona) [reply blockmanager.go:2623]: Could you assist?  I do not have a working dev env.

File: blockmanager.go (hunk 2 above)
Comment (cubancorona) [reply blockmanager.go:2625]: Could you assist?  I agree and many thanks for the review.

File: blockmanager.go (hunk 1 above)
Comment (guggero) [reply blockmanager.go:2623]: What do you mean by assist? Take over the PR? I can't push changes to your branch...

Comment (cubancorona) [reply blockmanager.go:2623]: Yes, that's fine!

**Comment by guggero** [comment]
Replaced by https://github.com/lightninglabs/neutrino/pull/272.

**Pull request closed without merging**